common/domain/replicationTaskExecutor_test.go

```diff
@@ -40,28 +42,28 @@ import (
 
 type (
 	domainReplicationTaskExecutorSuite struct {
-		suite.Suite
-		persistencetests.TestBase
+		*persistencetests.TestBase
 		domainReplicator *domainReplicationTaskExecutorImpl
 	}
 )
 
 func TestDomainReplicationTaskExecutorSuite(t *testing.T) {
 	testflags.RequireCassandra(t)
-	s := new(domainReplicationTaskExecutorSuite)
-	suite.Run(t, s)
-}
 
-func (s *domainReplicationTaskExecutorSuite) SetupSuite() {
-}
+	if testing.Verbose() {
+		log.SetOutput(os.Stdout)
+	}
```

Comment:
does using `t.Log`-proxied stuff work, or do these run into races if you do that?  I know it's not directly `log` compatible, but I doubt that matters in practice for our tests.

---

otherwise:
- default stderr -> verbose stdout: essentially matches `go test -v` (stdout) so 👍 that seems reasonable.
- to match `-v` more closely, do we want to maybe `io.Discard` when not verbose?  none of our existing stuff treats stdout vs stderr differently, and CI is verbose, so I don't know if there's much benefit to specializing it like this unless it's going to lead to a pragmatic or technical improvement somewhere.
- this seems broadly useful, given all the bare log use, and this one test doesn't seem special at all at a glance (is it in practice?).  seems worth adding to a test-helper package and using everywhere?

Reply:
In the following PR I want to move all logging into the root suite TestBase/IntegrationBase part and use -v to either use zaptest or use zap.NewDevelopment depending on the `-v`. At least that is my plan.

Reply:
if zaptest doesn't cause data races (due to unclean shutdown), _always_ prefer zaptest.  there's basically no reason to use zap.NewDevelopment or log.* or fmt.Println in tests, t.Log-based stuff is massively better.